src/main/scala/nl.knaw.dans.easy.bag2deposit/EasyConvertBagToDepositApp.scala

```diff
@@ -91,29 +98,32 @@ class EasyConvertBagToDepositApp(configuration: Configuration) extends DebugEnha
       bag <- BagFacade.getBag(bagDir)
       mutableBagMetadata = bag.getMetadata
       bagInfo <- BagInfo(bagDir, mutableBagMetadata)
-      _ = bagInfoKeysToRemove.foreach(mutableBagMetadata.remove)
       _ = logger.info(s"$bagInfo")
       metadata = bagDir / "metadata"
       migration = bagDir / "data" / "easy-migration"
       ddmFile = metadata / "dataset.xml"
       ddmIn <- loadXml(ddmFile)
-      props <- depositPropertiesFactory.create(bagInfo, ddmIn)
-      fromVault = props.getString("deposit.origin") == "VAULT"
-      datasetId = props.getString("identifier.fedora", "")
+      depositProps <- depositPropertiesFactory.create(bagInfo, ddmIn)
+      fromVault = depositProps.getString("deposit.origin") == "VAULT"
+      datasetId = depositProps.getString("identifier.fedora", "")
       ddmOut <- configuration.ddmTransformer.transform(ddmIn, datasetId)
       _ = registerMatchedReports(datasetId, ddmOut \\ "reportNumber")
-      _ = ddmFile.writeText(ddmOut.serialize)
       oldDcmi = (ddmIn \ "dcmiMetadata").headOption.getOrElse(<dcmiMetadata/>)
       newDcmi = (ddmOut \ "dcmiMetadata").headOption.getOrElse(<dcmiMetadata/>)
-      _ = props.save((bagParentDir / "deposit.properties").toJava) // N.B. the first write action
-      _ = if (fromVault) addAmdXml(datasetId, metadata)
-      amdChanges <- configuration.userTransformer.transform(metadata / "amd.xml")
-      _ = provenance.collectChangesInXmls(Map(
-        "http://easy.dans.knaw.nl/easy/dataset-administrative-metadata/" -> amdChanges,
-        "http://easy.dans.knaw.nl/schemas/md/ddm/" -> compare(oldDcmi, newDcmi),
-      )).foreach(xml => (metadata / "provenance.xml").writeText(xml.serialize))
-      _ = copyMigrationFiles(metadata, migration, fromVault)
+      amdFile = metadata / "amd.xml"
```

Comment:
the amd.xml has already been put here in case of VAULT bags?

Reply:
The next line tries to read the amd from the bag, if not found it reads from fedora. The `userTransformer` _may_ change it, once all changes are collected in memory, the amd is (re)written anyway.